great_expectations/expectations/expectation.py

```diff
@@ -3238,6 +3246,10 @@ def _format_map_output(
         return_obj["result"].update({"unexpected_index_list": unexpected_index_list})
     if unexpected_index_query is not None:
         return_obj["result"].update({"unexpected_index_query": unexpected_index_query})
+    if unexpected_index_column_names is not None:
+        return_obj["result"].update(
+            {"unexpected_index_column_names": unexpected_index_column_names}
+        )
```

Comment:
Ensuring `unexpected_index_column_names` is part of the Validation result object.